swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/SystemExtensionManager.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  SystemExtensionManager.swift
+//  FirezoneKit
+//
+//  Created by Jamil Bou Kheir on 11/15/24.
+//
+
+#if os(macOS)
+import SystemExtensions
+
+enum SystemExtensionError: Error {
+  case UnexpectedResult(result: OSSystemExtensionRequest.Result)
+  case NeedsUserApproval
+}
+
+public class SystemExtensionManager: NSObject, OSSystemExtensionRequestDelegate {
+  private var completionHandler: ((Error?) -> Void)?
+
+  public func installSystemExtension(
+    identifier: String?,
+    completionHandler: @escaping (Error?) -> Void
+  ) {
+    guard let identifier = identifier else { return }
+    self.completionHandler = completionHandler
+
+    let request = OSSystemExtensionRequest.activationRequest(forExtensionWithIdentifier: identifier, queue: .main)
+    request.delegate = self
+
+    // Install extension
+    OSSystemExtensionManager.shared.submitRequest(request)
+  }
+
+  // MARK: - OSSystemExtensionRequestDelegate
+
+  public func request(_ request: OSSystemExtensionRequest, didFinishWithResult result: OSSystemExtensionRequest.Result) {
+    guard result == .completed else {
+      completionHandler?(SystemExtensionError.UnexpectedResult(result: result))
+
+      return
+    }
+
+    // Success
+    completionHandler?(nil)
+  }
+
+  public func request(_ request: OSSystemExtensionRequest, didFailWithError error: Error) {
+    completionHandler?(error)
+  }
+
+  public func requestNeedsUserApproval(_ request: OSSystemExtensionRequest) {
+    completionHandler?(SystemExtensionError.NeedsUserApproval)
+
+    // TODO: Inform the user to approve the system extension in System Preferences > Security & Privacy.
```

Comment:
Does this need addressing?

Reply:
Not strictly required - I _think_ the user is shown a macOS alert when the sysex is installed, so this might not be necessary.

We may need to handle the edge case where the user dismisses it, then tries to sign in anyway, so this TODO might not be in the relevant place here. Will circle back to it when I get the release / distributable build PR open.